### DISCUSSION TITLE
Storage elements in swmmio

### DISCUSSION THREAD

**Discussion by bowguy** [pyswmm/swmmio#225] [Q&A] [answered]
I cannot get the correct information for storage elements in swmmio.  Using the example:

_from swmmio import Model, Nodes
m = Model("coolest-model.inp")

# pass custom init arguments into the Nodes object instead of using default settings referenced by m.nodes() 
nodes = Nodes(
    model=m, 
    inp_sections=['junctions', 'storages', 'outfalls'],
    rpt_sections=['Node Depth Summary', 'Node Inflow Summary'],
    columns=[ 'InvertElev', 'MaxDepth', 'InitDepth', 'SurchargeDepth', 'MaxTotalInflow', 'coords']
)
#write to a csv
nodes.to_csv('/path/mynodes.csv')

The resultant mynodes.csv has no references to any storage notes.  You must change the line:
    inp_sections=['junctions', 'storages', 'outfalls'],
to
    inp_sections=['junctions', 'storage', 'outfalls'],

After that change, the file has some numbers for the storage nodes but only for InverElev and coords.

**Comment by aerispaha** [comment, marked as answer]
Hey @bowguy! First, thanks for identifying an issue in the docs. The reason you aren't seeing any references to the storage elements in your model when calling `Nodes()` with the argument `inp_sections=['junctions', 'storages', 'outfalls']` is because, as you correctly identified, `storages` should in fact be `storage`. 

Now, after using the correct reference to `storage`, the reason you're only seeing data for the `InverElev` and `coords` is because of the list passed to the optional `columns` argument. The `columns` argument is used to slice out a subset of columns that would otherwise be returned by the `Nodes` object. This is useful if you know you need data from several sections of INPs and RTPs, but don't need _everything_ from those sections.

That said, if you do want all of the columns from all INP and RPT sections passed when initializing the Nodes object, you can simply do this:

```python
nodes = Nodes(
    model=model, 
    inp_sections=['junctions', 'storage', 'outfalls'],
    rpt_sections=['Node Depth Summary', 'Node Inflow Summary'],
)

nodes.dataframe
```

```
      InvertElev  MaxDepth  InitDepth  SurchargeDepth  PondedArea  MaxD StorageCurve  Coefficient  Exponent  Constant  EvapFrac OutfallType StageOrTimeseries Type_Node_Depth_Summary  AvgDepth  MaxNodeDepth  MaxHGL  MaxDay_Node_Depth_Summary MaxHr_Node_Depth_Summary  MaxNodeDepthReported      Type  MaxLatInflow  MaxTotalInflow  MaxDay  MaxHr  LatInflowV  TotalInflowV  FlowBalErrorPerc  XXX        X        Y                 coords
Name                                                                                                                                                                                                                                                                                                                                                                                                                                            
J3         6.547      15.0        0.0             0.0         0.0   NaN          NaN          NaN       NaN       NaN       NaN         NaN               NaN                JUNCTION      0.96          1.64    8.19                          0                    10:00                   NaN  JUNCTION          2.71            7.04       0  09:59       2.017         6.096               NaN  NaN  459.058 -113.145  [(459.058, -113.145)]
1         17.000       0.0        0.0             0.0         0.0   NaN          NaN          NaN       NaN       NaN       NaN         NaN               NaN                JUNCTION      0.07          0.94   17.94                          0                    10:00                   NaN  JUNCTION          2.61            2.61       0  09:59       0.131         0.131               NaN  NaN  -77.021  -78.321   [(-77.021, -78.321)]
2         17.000       0.0        0.0             0.0         0.0   NaN          NaN          NaN       NaN       NaN       NaN         NaN               NaN                JUNCTION      0.00          0.00   17.00                          0                    00:00                   NaN  JUNCTION          0.00            0.00       0  00:00       0.000         0.000               NaN  NaN  -84.988   43.833    [(-84.988, 43.833)]
3         16.500       0.0        0.0             0.0         0.0   NaN          NaN          NaN       NaN       NaN       NaN         NaN               NaN                JUNCTION      0.00          0.39   16.89                          0                    10:03                   NaN  JUNCTION          0.00            0.13       0  09:54       0.000         0.001               NaN  NaN  -18.600  -71.239     [(-18.6, -71.239)]
4         16.000       0.0        0.0             0.0         0.0   NaN          NaN          NaN       NaN       NaN       NaN         NaN               NaN                JUNCTION      0.07          0.87   16.87                          0                    10:02                   NaN  JUNCTION          0.00            2.54       0  10:00       0.000         0.131               NaN  NaN  -67.284  -37.603   [(-67.284, -37.603)]
5         15.000       0.0        0.0             0.0         0.0   NaN          NaN          NaN       NaN       NaN       NaN         NaN               NaN                JUNCTION      0.06          1.00   16.00                          0                    10:09                   NaN  JUNCTION          0.00            2.38       0  10:03       0.000         0.131               NaN  NaN  -56.662   15.507    [(-56.662, 15.507)]
J2        13.000      15.0        0.0             0.0         0.0   NaN          NaN          NaN       NaN       NaN       NaN         NaN               NaN                JUNCTION      0.00          0.00   13.00                          0                    00:00                   NaN  JUNCTION          2.19            4.33       0  09:59       1.991         4.078               NaN  NaN  238.750  -53.332    [(238.75, -53.332)]
J1        13.392       NaN        0.0             NaN         0.0  15.0   FUNCTIONAL       1000.0       0.0       0.0       0.0         NaN               NaN                 STORAGE      0.70          2.18   15.57                          0                    10:19                   NaN   STORAGE          1.55            3.52       0  09:59       1.962         2.093               NaN  NaN    0.000    0.000           [(0.0, 0.0)]
J4         0.000       NaN        NaN             NaN         NaN   NaN          NaN          NaN       NaN       NaN       NaN        FREE                NO                 OUTFALL      0.00          0.00    0.00                          0                    00:00                   NaN   OUTFALL          0.00            7.00       0  10:00       0.000         6.094               NaN  NaN  671.391 -163.985  [(671.391, -163.985)]
```

I'll fix the docs ASAP to correct the `storages` typo - thanks again for identifying that. 

Aside from that, will adjusting or omitting the `columns` argument address your issue?

**Reply by bowguy** [reply]
Thanks!  This did solve my issue on the nodes.  Still struggling with the storage nodes but that is another issue

**Reply by bowguy** [reply]
I need to get the following information about storage elements into a geojson file:
MaxDepth and MaxHGL from the 'Node Depth Summary' report section,
MaxTotalInflow from the 'Node Inflow Summary' report section and
MaxOutflow from the 'Storage Volume Summary' report section.
However, each of those report sections have MaxDay and MaxHr columns so combining any two gives the pandas error:

ValueError: columns overlap but no suffix specified: Index(['MaxDay'], dtype='object')

Is there anyway to get around this? I think I need to specify a suffix but not sure how to do this within the swmmio.elements.Nodes element.

**Reply by bowguy** [reply]
Actually, 'Node Inflow Summary' and 'Node Depth Summary' don't conflict, only when you add the 'Storage Volume Summary' do you get the conflict.  Where are these handled?
